engine/src/flutter/impeller/geometry/rational.cc

```diff
@@ -25,10 +25,6 @@ bool Rational::operator==(const Rational& that) const {
   }
 }
 
-bool Rational::operator!=(const Rational& that) const {
```

Comment:
Should this have spaceship? It defines `<` below...

Reply:
That would also add `>`, `>=`, and `<=`. Besides, the implementation seemed to do custom things. I wanted to keep the behavior the same in this patch. But if you want to add it now, we should totally go for it.

Reply:
Does spaceship have a "NaN" return value? I guess since it can return any comparable value, it could return a scalar and that could be NaN...

This would come up with denominators of 0.

BTW, it occurs to me that the implementation of these methods doesn't deal with 0 denominators anyway?